common/certificate/src/validate_device_id.rs

```diff
@@ -0,0 +1,75 @@
+use crate::CertificateError;
+
+fn is_valid_device_id_char(ch: char) -> bool {
+    matches!(ch, 'A'..='Z'
+    | 'a'..='z'
+    | '0'..='9'
+    | '\''
+    | '='
+    | '('
+    | ')'
+    | '+'
```

Comment:
Do we want to allow single character names, especially if those characters are special characters like `+` that has special meaning in the MQTT world?

Reply:
Good point, I agree we should avoid such things. @didier-wenzek  can pitch in here. Then we should come up with comprehensive naming conventions than this one. Like enforcing the name to have minimum length etc.